src/lib/browse/CensusOutputAreaLayerControl.svelte

```diff
@@ -0,0 +1,244 @@
+      colorBy = "percent_households_without_car";
+    } else if (showAverageCars) {
+      colorBy = "average_cars_per_household";
+    } else if (showPopulationDensity) {
+      colorBy = "pop_density";
+    } else {
+      colorBy = "";
+    }
+    if (colorBy) {
+      $map.setPaintProperty(name, "fill-color", makeColorRamp());
+      // InteractiveLayer manages the polygon layer, but we also need to control the outline
+      $map.setLayoutProperty(outlineLayer, "visibility", "visible");
+    } else {
+      $map.setLayoutProperty(outlineLayer, "visibility", "none");
+    }
+  }
+
+  overwritePmtilesSource(
+    $map,
+    name,
+    `https://atip.uk/layers/v1/${name}.pmtiles`
+  );
+
+  overwritePolygonLayer($map, {
+    id: name,
+    source: name,
+    sourceLayer: name,
+    // Initially set to a dummy value
+    color: "black",
+    opacity: hoveredToggle(0.5, 0.7),
+  });
+  overwriteLineLayer($map, {
+    id: outlineLayer,
+    source: name,
+    sourceLayer: name,
+    color: "black",
+    width: 0.5,
+  });
+
+  function tooltip(feature: MapGeoJSONFeature): string {
+    let oa = feature.properties["OA21CD"];
+    let value = feature.properties[colorBy];
+    if (colorBy == "percent_households_without_car") {
+      return `<p>${value}% of households in ${oa} don't have a car</p>`;
+    } else if (colorBy == "average_cars_per_household") {
+      return `<p>Households in ${oa} have an average of ${value} cars</p>`;
+    } else {
+      return `<p>There are ${value.toLocaleString()} people per square kilometre in ${oa}</p>`;
+    }
+  }
+
+  function onClick(e: CustomEvent<MapGeoJSONFeature>) {
+    let oa = e.detail.properties["OA21CD"];
+    if (
+      colorBy == "percent_households_without_car" ||
+      colorBy == "average_cars_per_household"
+    ) {
+      window.open(
+        `https://www.ons.gov.uk/census/maps/choropleth/housing/number-of-cars-or-vans/number-of-cars-5a/no-cars-or-vans-in-household?oa=${oa}`,
+        "_blank"
+      );
+    } else {
+      window.open(
+        `https://www.ons.gov.uk/census/maps/choropleth/population/population-density/population-density/persons-per-square-kilometre?oa=${oa}`,
+        "_blank"
+      );
+    }
+  }
+
+  // Should return 6 things (1 more than colorScale)
+  function makeLimits(): number[] {
+    if (colorBy == "percent_households_without_car") {
+      // Equal buckets of 20%
+      return [0, 20, 40, 60, 80, 100];
+    } else if (colorBy == "average_cars_per_household") {
+      // The max value is 2.4, so pick equal buckets
+      return [0.0, 0.5, 1.0, 1.5, 2.0, 2.5];
+    } else {
+      // Use the same (slightly rounded) buckets as https://www.ons.gov.uk/census/maps/choropleth/population/population-density/population-density/persons-per-square-kilometre
+      return [0, 4700, 13000, 33000, 94000, 1980000];
+    }
+  }
+
+  function makeColorRamp(): any[] {
+    let limits = makeLimits();
+    let fillColor: any[] = ["step", ["get", colorBy]];
```

Comment:
See https://maplibre.org/maplibre-style-spec/expressions/#step for reference... and note the docs don't match reality or the example code, because `stop_output_0` is a lie. Instead there's a fallback value at the very end. I'll file an issue / send a PR to fix the docs...